muikku-core-plugins/src/main/java/fi/otavanopisto/muikku/plugins/search/CommunicatorMessageIndexer.java

```diff
@@ -0,0 +1,147 @@
+      for (CommunicatorMessageRecipient recipient : recipientsList) {
+        Long recipientId = recipient.getRecipient();
+        UserEntity recipientEntity = userEntityController.findUserEntityById(recipientId);
+        UserEntityName recipientName = userEntityController.getName(recipientEntity);
+
+        if (recipientName != null) {
+          IndexedCommunicatorMessageRecipient recipientData = new IndexedCommunicatorMessageRecipient();
+          
+          recipientData.setUserEntityId(recipientId);
+          recipientData.setFirstName(recipientName.getFirstName());
+          recipientData.setLastName(recipientName.getLastName());
+          recipientData.setNickName(recipientName.getNickName());
+          recipientData.setStudyProgrammeName(recipientName.getStudyProgrammeName());
+          // set is message read/unread by receiver
+          recipientData.setReadByReceiver(recipient.getReadByReceiver());
+          recipientData.setArchivedByReceiver(recipient.getArchivedByReceiver());
+            
+          // set labels
+            List<IndexedCommunicatorMessageLabels> labelsList = new ArrayList<IndexedCommunicatorMessageLabels>();
+            List<CommunicatorMessageIdLabel> labels = communicatorMessageIdLabelDAO.listByUserAndMessageId(recipientEntity, communicatorMessageId);
+            for (CommunicatorMessageIdLabel label : labels) {
+              IndexedCommunicatorMessageLabels labelData = new IndexedCommunicatorMessageLabels();
+              CommunicatorLabel wholeLabel = label.getLabel();
+              
+              labelData.setLabel(wholeLabel.getName());
+              labelData.setId(wholeLabel.getId());
+              labelsList.add(labelData);
+            } 
+              
+            recipientData.setLabels(labelsList);
+            recipientsEntityList.add(recipientData);
+          }
+        }
+	    	
+        indexedCommunicatorMessage.setRecipients(recipientsEntityList);
+	    	
+        // set created
+        Date created = message.getCreated();
+        indexedCommunicatorMessage.setCreated(created);
+	    	
+        indexedCommunicatorMessage.setSearchId(message.getId());
+	        
+        //call method indexCommunicatorMessage
+      indexCommunicatorMessage(indexedCommunicatorMessage);
+    } else {
+      logger.warning(String.format("could not index communicator message because message entity #%s/ %s could not be found", message));
+    }
+  }
+  
+  private void indexCommunicatorMessage(IndexedCommunicatorMessage indexedCommunicatorMessage) {
+    try {
+      if (indexedCommunicatorMessage != null) {
+        indexer.index(indexedCommunicatorMessage.getClass().getSimpleName(), indexedCommunicatorMessage);
+      }
+    } catch (Exception e) {
+      logger.warning(String.format("could not index message #%s/%s//%s", indexedCommunicatorMessage.getClass().getSimpleName(), indexedCommunicatorMessage.getId(), e));
```

Comment:
Exception to string might not be informative enough? Maybe logger.log(Level.WARNING, "msg", e); instead?